course/utils.py

```diff
@@ -659,7 +660,19 @@ def has_permission(self, perm, argument=None):
         else:
             return (perm, argument) in self.permissions()
 
+    def _set_course_lang(self, action):
+        # type: (Text) -> None
+        from django.conf import settings
+        if not getattr(settings, "RELATE_ENABLE_COURSE_SPECIFIC_LANG", False):
+            return
+        if self.course.force_lang is not None:
+            if action == "activate":
+                translation.activate(self.course.force_lang)
+            else:
+                translation.deactivate()
+
     def __enter__(self):
+        self._set_course_lang(action="activate")
         return self
 
     def __exit__(self, exc_type, exc_val, exc_tb):
```

Comment:
How come this makes no effort to restore the prior state on `__exit__`?

Reply:
Thanks, I'll try that.

Reply:
Sorry, I did do the restoring but I didn't know why that line just disappear...

Reply:
Updated.